app/controllers/application_records_controller.rb

```diff
@@ -66,6 +66,22 @@ def review
     show_message :application_review,
                  default: 'Application has been marked as reviewed.'
     @record.update reviewed: true
+    @record.update saved_for_later: false
+    redirect_to staff_dashboard_path
+  end
+
+  def toggle_saved_for_later
+    if @record.saved_for_later
+      @record.unsave
+      flash[:message] = 'Application moved back to dashboard.'
+    else
+      date = if params[:date_for_later].present?
```

Comment:
No, no problems if date for later is nil. It is not called when it could be nil.